src/main/java/org/jboss/classfilewriter/ClassFile.java

```diff
@@ -104,6 +105,27 @@ public ClassFile(String name, int accessFlags, String superclass, int version, C
         this.attributes.add(runtimeVisibleAnnotationsAttribute);
     }
 
+    private static java.lang.reflect.Method defineClass1, defineClass2;
+
+    static {
+        try {
+            AccessController.doPrivileged(new PrivilegedExceptionAction<Object>() {
+                public Object run() throws Exception {
+                    Class<?> cl = Class.forName("java.lang.ClassLoader", false, null);
+                    defineClass1 = cl.getDeclaredMethod("defineClass", new Class[] { String.class, byte[].class, int.class,
+                            int.class });
+                    defineClass1.setAccessible(true);
```

Comment:
This will warn/fail.  I think the point of using Unsafe was to not call setAccessible.  So probably my earlier comments should be disregarded: the declared methods should be acquired in the privileged block, then their "override" flags should be toggled by Unsafe, and finally the resultant Method object(s) can be cached in final fields with no further usage of Unsafe being necessary.

Reply:
Oops this is a copy-paste oversight (took this bit of code from earlier state of repo). Thanks for pointing this out.